### PR TITLE
[BUGFIX] Patch additional issues with data docs page retrieval in checkpoint actions

### PR DESCRIPTION
**Pull request by cdkini**
Continuation of https://github.com/great-expectations/great_expectations/pull/10374

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/render/renderer/renderer.py [hunk 1]
```diff
@@ -37,6 +38,10 @@ def inner_func(*args: P.args, **kwargs: P.kwargs):
 class Renderer:
     """A convenience class to provide an explicit mechanism to instantiate any Renderer."""
 
+    @override
+    def __eq__(self, other: object) -> bool:
+        return type(self) is type(other)
```
Comment (billdirks) [inline great_expectations/render/renderer/renderer.py:44]: Is this check good enough for renderers? We don't need to check anything inside them?

**Review by billdirks — COMMENTED** [review]
Thanks for doing this. I don't know all the details of the issue surrounding this, but the changes mostly look good to me. Is checking type enough for equality for renderers?

**Review by tyler-hoffman — APPROVED** [review]

File: great_expectations/render/renderer/renderer.py (hunk 1 above)
Comment (billdirks) [reply great_expectations/render/renderer/renderer.py:44]: Chatted offline. Renderers have so state so this is ok for equality.

**Review by billdirks — APPROVED** [review]
Thanks!

**Pull request merged**